cmd/opampsupervisor/supervisor/config/config.go

```diff
@@ -121,6 +121,8 @@ type Agent struct {
 	Executable              string
 	OrphanDetectionInterval time.Duration    `mapstructure:"orphan_detection_interval"`
 	Description             AgentDescription `mapstructure:"description"`
+	SuccessfulHealthChecks  int32            `mapstructure:"successful_health_checks"`
+	ConfigApplyTimeout      time.Duration    `mapstructure:"config_apply_timeout"`
```

Comment:
I've been thinking about this a bit, wondering if maybe the timeout should actually be something like the max number of checks to do instead of a duration.

Basically, the issue with this configuration that I'm seeing is that it's a little hard to tell the relationship between `successful_health_checks` and `config_apply_timeout`. Like, it's possible to set `successful_health_checks` to a value that is large enough that `config_apply_timeout` will always fail, but it's not obvious what the limit is here.

Is there a way we can protect against that somehow?

Reply:
> wondering if maybe the timeout should actually be something like the max number of checks to do instead of a duration.

can you elaborate on this a bit? When you say the max number of checks, do you mean the maximum number of regardless of the status of the health check? 

One of the reasons to have both settings is when the first health check may succeed, but subsequent ones could fail because of some async error, so it made sense to add consecutive health checks to pass before considering the collector is healthy. The timeout exists to break out of the loop in case it keeps restarting. Can clarifying how health check ticks are performed help?

Reply:
Sorry, let me restate the problem.

I am a user configuring this. In order to understand what values are valid for both of these, I need to know how many health checks happen within a given period of time.

For instance, `config_apply_timeout` is 30 seconds by default. How many health checks occurs in this period? If I don't know this information, I don't know what the limit for `successful_health_checks` is.

The two ways I see to fix that are:
1. Keep the current options from this pr, add and option for configuring the health check interval.
2. Alternatively, specify the number of health checks, total, that will performed before giving up.

I do think having the timeout be an actual time interval makes sense, I just want some way to correlate it with how it relates to the `successful_health_checks`.

Reply:
>Alternatively, specify the number of health checks, total, that will performed before giving up.

With this option what is the criteria for deciding if the collector is healthy?

Reply:
It would be the same criteria as you have now (e.g. if `successful_health_checks` is 3, then it would wait for 3 successful health checks to mark the collector as healthy). 

The difference is that instead of a timeout interval, you'd specify the number of health checks to do, total. So if a new param, say `config_apply_health_checks`, is 5 - that means that up to 5 health checks will be performed before failing the apply operation if not healthy. The point being that I know that since 3 < 5,  `successful_health_checks` set to 3 would be valid here.

Reply:
Thanks, now I understand your point. I am inclined to add the health check interval option as it would be useful in general. What do you think?

Reply:
Yeah, that sounds like a good solution.

Reply:
Added the `health_check_interval` config param to agent config.